core/templates/dev/head/pages/library/Library.js

```diff
@@ -55,7 +58,11 @@ oppia.controller('Library', [
 
     $scope.activeGroupIndex = null;
 
-    $scope.pageMode = GLOBALS.PAGE_MODE;
+    var currentPath = $window.location.pathname;
+    $scope.pageMode = LIBRARY_PATHS_AND_MODES[currentPath];
+    if (!$scope.pageMode) {
+      $log.error('INVALID URL PATH: ' + currentPath);
```

Comment:
I think we use `throw Error(..)` pattern.

Reply:
I found `$log.error` being used [here](https://github.com/oppia/oppia/blob/develop/core/templates/dev/head/pages/library/Library.js#L108) in Library.js

Reply:
I think throw Error will stop the execution, do we want to continue the execution if the URL is invalid?

Also, I have a question: with the invalid path, we will have a 404 page right?

Reply:
Yeah, in case of invalid path there will be a 404 page. So, `scope.pageMode` will not be executed in that case. I added it because it was useful while debugging. It can be useful when `currentPath` is not present in  LIBRARY_PATHS_AND_MODES while making changes to code. Should i remove it?